File: Courses/Basic Deutsch/course.py
import Courses

    # Course Details
COURSE = Courses.course(
    TITLE= "Basic Deutsch",
    SETTINGS= SETTINGS,
    LESSONS= LESSONS)

    # Course Settings
SETTINGS = Courses.settings(
    API= Courses.API(
        TRANSLATOR= "libre",    # Preferred Translator API
        TTS= "mozilla",         # Preferred Text-to-Speech API
        STT= "mozilla"          # Preferred Speech-to-Text API
        ),

    LANG= Courses.lang(
        NATIVE= "en",   # Two-letter Abbreviation for the Native Language
        TARGET= "de"    # Two-letter Abbreviation for the Target Language
        ))

    # Course Lessons
LESSONS = [
    Courses.lesson(
        TITLE= "Basics 1", 
        ICON = "check", 
        CHECKPOINT= 0, 
        LEVEL= 0, 
        MEDIA= Courses.media(
                # -NOTE- : All media but images are grouped together based on when they come in the list.
                # The first native sentence and audio must always be the translation of the first target sentence and audio, and so on.
            NATIVESENTENCES= [      # List of Native Sentences
                "Hello! I'm Melanie!"
                "What was there?"
            ],
            NATIVEAUDIO= [          # List of Native Audio File Paths
                # Enter Paths Here
            ],
            TARGETSENTENCES= [      # List of Target Sentences
                "Hallo! Ich bin Melanie!"
                "Was war da?"
            ],
            TARGETAUDIO= [          # List of Audio File Paths
                # Enter Paths Here
            ],
            IMAGES= [               # List of Image Classes
                # Enter Images Here
                """
                Courses.image(
                    PATH= "./images/apple.png", 
                    NATIVETEXT= "Apple",
                    TARGETTEXT= "Apfel"
                )
                """
            ]))
]

File: Courses/Basic Deutsch/course.mjs
import * as Courses from "../Courses.mjs"

const TITLE = ""  // Course Title

    // Course Settings
const SETTINGS = new Courses.settings(
    new Courses.API(
        "",     // Preferred Translator API
        "",     // Preferred Text-to-Speech API
        ""      // Preferred Speech-to-Text API
    ),

    new Courses.lang(
        "aa",   // Two-letter Abbreviation for the Native Language
        "bb"    // Two-letter Abbreviation for the Target Language
    )
)

    // Course Lessons
const LESSONS = [
        // -Lesson 0-
    new Courses.lesson(
        "",                 // Title
        "",                 // Icon Name 
        0,                  // Checkpoint Level (DEFAULT=0) 
        0,                  // Level within Checkpoint (DEFAULT=0)
        new Courses.media(  // Media Object used in Lesson

        // -NOTE- : All media but images are grouped together based on when they come in the list.
        // The first native sentence and audio must always be the translation of the first target sentence and audio, and so on.
        
            /* List of Native Sentences */ [
    
                // Enter Sentences Here
    
            ],
    
            /* List of Native Audio File Paths */ [
    
                // Enter Paths Here
    
            ],
    
            /* List of Target Sentences */ [
    
                // Enter Sentences Here
    
            ],
    
            /* List of Audio File Paths */ [
    
                // Enter Paths Here
    
            ],
    
            /*List of Image Classes*/ [
    
                new Courses.image(
                    "", // Path to Image
                    "", // Native Description
                    ""  // Target Description
                )
            ]
        ) 
    )
]

    // Course Details
const COURSE = new Courses.course(
    TITLE,      // Course Title
    SETTINGS,   // Course Settings
    LESSONS     // Course Lessons
)

console.log(COURSE.TITLE)

export { COURSE }